pkg/config/pre_process_test.go

```diff
@@ -14,37 +14,47 @@
 
 package config
 
-/*func TestNewProcessorsAndRun(t *testing.T) {
-	config := &v1.Config{
-		Spec: v1.ConfigSpec{
-			Process: "value|toJson|toBase64|toSecret",
-			Data: `
+import (
+	"encoding/base64"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	v1 "github.com/sealerio/sealer/types/api/v1"
+)
+
+func TestNewProcessorsAndRun(t *testing.T) {
+	tests := []struct {
+		name     string
+		config   *v1.Config
+		wantData string
+		wantErr  bool
+	}{
+		{
+			name: "value|toJson|toBase64|toSecret",
+			config: &v1.Config{
+				Spec: v1.ConfigSpec{
+					Process: "value|toJson|toBase64|toSecret",
+					Data: `
 config:
   username: root
   passwd: xxx
 `,
+				},
+			},
+			wantData: "config: " + base64.StdEncoding.EncodeToString([]byte(`{"passwd":"xxx","username":"root"}`)) + "\n",
+			wantErr:  false,
 		},
 	}
 
-	type args struct {
-		config *v1.Config
-	}
-	tests := []struct {
-		name    string
-		args    args
-		wantErr bool
-	}{
-		{
-			name:    "test value|toJson|toBase64|toSecret",
-			args:    args{config},
-			wantErr: false,
-		},
-	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if err := NewProcessorsAndRun(tt.args.config); (err != nil) != tt.wantErr || tt.args.config.Spec.Data != "config: eyJwYXNzd2QiOiJ4eHgiLCJ1c3JuYW1lIjoicm9vdCJ9\n" {
+			if err := NewProcessorsAndRun(tt.config); (err != nil) != tt.wantErr {
 				t.Errorf("NewProcessorsAndRun() error = %v, wantErr %v", err, tt.wantErr)
 			}
+			if diff := cmp.Diff(tt.config.Spec.Data, tt.wantData); diff != "" {
```

Comment:
recommend using "assert.Equal(t, tt.config.Spec.Data, tt.wantData)", it will give us more details.

Reply:
Thanks for your suggestion, sure I will use assert